Mathlib/Algebra/Ring/Basic.lean

```diff
@@ -35,22 +35,34 @@ instance (priority := low) [HasNumerals α] : OfNat α n where
 # Semirings and rings
 -/
 
-class Semiring (R : Type u) extends AddCommMonoid R, MonoidWithZero R, HasNumerals R where
-  mul_add (a b c : R) : a * (b + c) = a * b + a * c
-  add_mul (a b c : R) : (a + b) * c = a * c + b * c
+/-- A typeclass stating that multiplication is left and right distributive
+over addition. -/
+class Distrib (R : Type u) extends Mul R, Add R where
```

Comment:
Eh, return of the `Distrib`. I'd rather we not have this / backport remove it; do we actually need it? Also the names `left_distrib` and `right_distrib` are objectively worse than `mul_add` and `add_mul`.

Reply:
I don't think it's a good idea to do that refactoring as part of the port.  Either now in mathlib3 or later once the port is finished.  (I hope I'm not missing an open mathlib PR or something? https://leanprover-community.github.io/mathlib_docs/algebra/ring/basic.html#distrib)

The motivation of this PR is---as the title says---to get the mathlib4 hierarchy closer to what we have in mathlib3.  This has already uncovered a few bugs in the structure command so it's certainly useful.

PS: I don't like the `left_distrib` names either.